nix/haskell.nix

```diff
@@ -0,0 +1,149 @@
+    } + dir;
+
+  pkgSet = haskell.mkStackPkgSet {
+    inherit stack-pkgs;
+    modules = [
+      # Add source filtering to local packages
+      {
+        packages.cardano-wallet-core.src = filterSubDir /lib/core;
+        packages.cardano-wallet-core.components.tests.unit.keepSource = true;
+        packages.cardano-wallet-core-integration.src = filterSubDir /lib/core-integration;
+        packages.cardano-wallet-cli.src = filterSubDir /lib/cli;
+        packages.cardano-wallet-launcher.src = filterSubDir /lib/launcher;
+        packages.cardano-wallet-byron.src = filterSubDir /lib/byron;
+        packages.cardano-wallet-jormungandr.src = filterSubDir /lib/jormungandr;
+        packages.cardano-wallet-jormungandr.components.tests.unit.keepSource = true;
+        packages.cardano-wallet-jormungandr.components.tests.integration.keepSource = true;
+        packages.cardano-wallet-test-utils.src = filterSubDir /lib/test-utils;
+        packages.text-class.src = filterSubDir /lib/text-class;
+        packages.text-class.components.tests.unit.keepSource = true;
+      }
+
+      # Add dependencies
+      {
+        packages.cardano-wallet-jormungandr.components.tests = {
+          # Some tests want to write ~/.local/share/cardano-wallet
+          integration.preCheck = "export HOME=`pwd`";
+          # provide jormungandr command to test suites
+          integration.build-tools = [
+            jmPkgs.jormungandr
+            jmPkgs.jormungandr-cli
+          ];
+          unit.build-tools = [ jmPkgs.jormungandr ];
+        };
+        packages.cardano-wallet-jormungandr.components.benchmarks.latency =
+          pkgs.lib.optionalAttrs (!pkgs.stdenv.hostPlatform.isWindows) {
+            build-tools = [ pkgs.makeWrapper];
+            postInstall = ''
+              wrapProgram $out/bin/latency \
+                --run "cd $src" \
+                --prefix PATH : ${jmPkgs.jormungandr}/bin
+            '';
+          };
+
+        packages.cardano-wallet-core.components.tests.unit.preBuild = ''
+          export SWAGGER_YAML=${src + /specifications/api/swagger.yaml}
+        '';
+
+        # Workaround for Haskell.nix issue
+        packages.cardano-wallet-jormungandr.components.all.postInstall = pkgs.lib.mkForce "";
+        packages.cardano-wallet-core.components.all.preBuild = pkgs.lib.mkForce "";
+      }
+
+      # Allow reinstallation of Win32
+      { nonReinstallablePkgs =
+        [ "rts" "ghc-heap" "ghc-prim" "integer-gmp" "integer-simple" "base"
+          "deepseq" "array" "ghc-boot-th" "pretty" "template-haskell"
+          # ghcjs custom packages
+          "ghcjs-prim" "ghcjs-th"
+          "ghc-boot"
+          "ghc" "array" "binary" "bytestring" "containers"
+          "filepath" "ghc-boot" "ghc-compact" "ghc-prim"
+          # "ghci" "haskeline"
+          "hpc"
+          "mtl" "parsec" "text" "transformers"
+          "xhtml"
+          # "stm" "terminfo"
+        ];
+        #reinstallableLibGhc = true;
+      }
```

Comment:
This will result in `configFiles` not found. Primarily because for windows we'll eventually end up depending on Win32, and the stackage snapshot does not contain the `Win32` package.